### PR TITLE
Branch comparison for correcting force push

### PR DESCRIPTION
**Pull request by jbrillon**
PR to review the merge commit to correct the force push on [PR111](https://github.com/dougshidong/PHiLiP/pull/111)

### CODE REVIEW

File: src/dg/dg.cpp [hunk 1]
```diff
@@ -2897,13 +2913,7 @@ void DGBase<dim,real,MeshType>::evaluate_mass_matrices (bool do_inverse_mass_mat
         if (use_auxiliary_eq){
             global_mass_matrix_auxiliary.compress(dealii::VectorOperation::insert);
         }
-        //std::cout << " global_mass_matrix "  << std::endl;
-        //std::cout << std::setprecision(std::numeric_limits<long double>::digits10 + 1);
-        //global_mass_matrix.print(std::cout);
-        ////std::abort();
     }
-
-    return;
```
Comment (AlexanderCicchino) [inline src/dg/dg.cpp:2906]: This return statement should be here. From Doug's master, might have something to do with linear solver.

File: tests/unit_tests/flow_variable_tests/auxiliary_variable_test.cpp [hunk 2]
```diff
@@ -84,273 +83,256 @@ int main (int argc, char * argv[])
...
+            using PDE_enum = Parameters::AllParameters::PartialDifferentialEquation;
+            using ODE_enum = Parameters::ODESolverParam::ODESolverEnum;
+            all_parameters_new.pde_type = PDE_enum::diffusion;
+            all_parameters_new.use_weak_form = false;
+            all_parameters_new.use_periodic_bc = true;
+            all_parameters_new.ode_solver_param.ode_solver_type = ODE_enum::explicit_solver;//auxiliary only works explicit for now
+            all_parameters_new.use_inverse_mass_on_the_fly = true;
+            std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim,double>::create_discontinuous_galerkin(&all_parameters_new, poly_degree, poly_degree, grid_degree, grid);
+            pcout<<"going in allocate"<<std::endl;
+            dg->allocate_system (false,false,false);
```
Comment (AlexanderCicchino) [inline tests/unit_tests/flow_variable_tests/auxiliary_variable_test.cpp:125]: Why did you give this all an extra tab?

File: src/physics/burgers.cpp [hunk 3]
```diff
@@ -89,19 +89,6 @@ real Burgers<dim,nstate,real>::convective_surface_numerical_split_flux (
...
-//                const std::array< dealii::Tensor<1,dim,real>, nstate > &flux_interp_to_surface) const
-//{
-//    std::array<dealii::Tensor<1,dim,real>,nstate> surface_split_flux;
-//        for (int idim=0; idim<dim; ++idim) {
-//            for (int s=0; s<nstate; ++s) {
-//                surface_split_flux[s][idim] = 2.0/3.0 * flux_interp_to_surface[s][idim] + 1.0/3.0 * surface_flux[s][idim];
-//            }
-//        }
-//    return surface_split_flux;
-//}
```
Comment (AlexanderCicchino) [inline src/physics/burgers.cpp:104]: You should leave this in because it might be used, as we discussed .

File: src/dg/strong_dg.hpp [hunk 4]
```diff
@@ -30,21 +31,21 @@ class DGStrong: public DGBaseState<dim, nstate, real, MeshType>
     /// Destructor
     ~DGStrong();
 
-    ///Allocates the auxiliary equations' variables and RHS.
+    /// Allocates the auxiliary equations' variables and RHS.
```
Comment (AlexanderCicchino) [inline src/dg/strong_dg.hpp:34]: There should be a space after the /// in header files. This is the convention that Doug has used throughout the code. Makes for cleaner doc.

Comment (AlexanderCicchino) [reply src/dg/strong_dg.hpp:34]: I read it backwards, this edit is correct, sorry.

File: src/dg/dg.cpp (hunk 1 above)
Comment (jbrillon) [reply src/dg/dg.cpp:2906]: It doesn't actually serve any purpose here

File: tests/unit_tests/flow_variable_tests/auxiliary_variable_test.cpp (hunk 2 above)
Comment (jbrillon) [reply tests/unit_tests/flow_variable_tests/auxiliary_variable_test.cpp:125]: there isn't an extra tab, see the full file, this is inside a for loop within a for loop

File: src/physics/burgers.cpp (hunk 3 above)
Comment (jbrillon) [reply src/physics/burgers.cpp:104]: `convective_surface_numerical_split_flux` with the commented return type and argument types was removed everywhere else in the `src/physics/` directory so I thought this was no longer relevant

File: src/dg/dg.cpp (hunk 1 above)
Comment (AlexanderCicchino) [reply src/dg/dg.cpp:2906]: I agree, was just looking for things that are different. I don't know where your bug is then.

File: src/dg/strong_dg.cpp [hunk 5]
```diff
@@ -642,7 +642,7 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_volume_term_strong(
                 vol_flux_node[idim] = metric_oper.flux_nodes_vol[idim][iquad];
             }
             //compute the physical source
-            source = this->pde_physics_double->source_term (vol_flux_node, soln_state, current_cell_index, this->current_time);
+            source = this->pde_physics_double->source_term (vol_flux_node, soln_state, this->current_time, current_cell_index);
```
Comment (AlexanderCicchino) [inline src/dg/strong_dg.cpp:645]: This should explain the convection diffusion issue too hopefully.

Comment (jbrillon) [reply src/dg/strong_dg.cpp:645]: yep it actually resolved the issue

**Comment by jbrillon** [comment]
You can review it now -- everything that should be passing is passing; I'll run ctest on your branch overnight to make sure the timings are the same

**Comment by AlexanderCicchino** [comment]
Post the ctest results when it's done of both branches and if they're basically the same then it's good to push

**Comment by jbrillon** [comment]
Sounds good, I will re-run both tonight and post the results sometime this weekend.

**Comment by AlexanderCicchino** [comment]
Perfect. I'm close to fixing a pretty big bug allowing us to use the steady state solver. There shouldn't be complicated merge-conflicts with it and I'll let you know when it's pushed.

**Comment by AlexanderCicchino** [comment]
[ctest_results_Aug_15_2022.log](https://github.com/AlexanderCicchino/PHiLiP/files/9352681/ctest_results_Aug_15_2022.log)

**Pull request merged**